modules/core/src/main/java/org/apache/synapse/mediators/builtin/RespondMediator.java

```diff
@@ -28,6 +31,14 @@ public boolean mediate(MessageContext synCtx) {
 
         synCtx.setTo(null);
         synCtx.setResponse(true);
+
+        // if this is not a response from a proxy service
+        String proxyName = (String) synCtx.getProperty(SynapseConstants.PROXY_SERVICE);
+        if (proxyName == null || "".equals(proxyName)) {
```

Comment:
Shall we use isEmpty() method to check empty string?

Reply:
Fixed, thanks